app/Main.hs

```diff
@@ -180,8 +190,10 @@ parseSecret line =
                 }
 
 
-readSecretList :: FilePath -> IO (Either String [Secret])
-readSecretList fname = fmap (sequence . fmap parseSecret . lines) (readFile fname)
+readSecretList :: (MonadError VaultError m, MonadIO m) => FilePath -> m [Secret]
+readSecretList fname = do
+  esecrets <- liftIO $ fmap (sequence . fmap parseSecret . lines) (readFile fname)
+  either (throwError . IOError) (return . id) esecrets
```

Comment:
This seems to be a parse error, not an I/O error.

Reply:
Why compose with `id`?

Reply:
yes you're right. I got confused because I was also thinking: what happens if `readFile` fails? I've taken into account that possibility in one of my next commits.